### PR TITLE
feat: add all-pods log flag to kubectl

### LINKED ISSUE #1520: Make `kubectl log deployment/foo` read logs from **all** pods
**What would you like to be added**:

Up to now, `kubectl log deployment/foo` outputs the logs of **one** pod:

```
❯ kubectl logs deployments/foo --follow >> foo.log
Found 2 pods, using pod/foo-647ffff4b6-k9c5c
```

It would be very handy if `kubectl` could output the logs of all pods of the deployment.

**Why is this needed**:

The workaround is to use label selectors:

```
kubectl logs -l app=myapp
```

But that's not easily automated.

Imagine you get the name and the namespace of a deployment in a script.

There is no easy and automated way to get the matching label selector.

It would be great if `kubectl` could support reading the logs from **all** pods of a deployment.

Related thread on slack: https://kubernetes.slack.com/archives/C2GL57FJ4/p1700558613585049?thread_ts=1700558613.585049&cid=C2GL57FJ4

### PR DESCRIPTION
**Pull request by cmwylie19**
#### What type of PR is this?

/kind feature

ref: https://github.com/kubernetes/kubernetes/pull/123874

#### What this PR does / why we need it:

This adds an `--all-pods` option to get all pod logs from replicaSets or services by looking at the label selector and finding corresponding pods in the namespace. It reduces the need to look for logs based on labels. 

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubectl/issues/1520

#### Does this PR introduce a user-facing change?:
```release-note
Added flag to `kubectl logs` called `--all-pods` to get all pods from a object that uses a pod selector.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```bash
kubectl logs <deploy/sts/daemonset/svc>/<name> --all-pods 
```

usage
```bash
┌─[cmwylie19@Cases-MacBook-Pro] - [~/kubernetes] - [2024-03-11 06:37:09]
└─[0] <git:(1520 570a1ede3fd✱) > go run cmd/kubectl/kubectl.go logs svc/y --all-pods
/docker-entrypoint.sh: /docker-entrypoint.d/ is not empty, will attempt to perform configuration
/docker-entrypoint.sh: Looking for shell scripts in /docker-entrypoint.d/
/docker-entrypoint.sh: Launching /docker-entrypoint.d/10-listen-on-ipv6-by-default.sh
10-listen-on-ipv6-by-default.sh: info: Getting the checksum of /etc/nginx/conf.d/default.conf
10-listen-on-ipv6-by-default.sh: info: Enabled listen on IPv6 in /etc/nginx/conf.d/default.conf
/docker-entrypoint.sh: Sourcing /docker-entrypoint.d/15-local-resolvers.envsh
/docker-entrypoint.sh: Launching /docker-entrypoint.d/20-envsubst-on-templates.sh
/docker-entrypoint.sh: Launching /docker-entrypoint.d/30-tune-worker-processes.sh
/docker-entrypoint.sh: Configuration complete; ready for start up
2024/03/07 21:10:16 [notice] 1#1: using the "epoll" event method
2024/03/07 21:10:16 [notice] 1#1: nginx/1.25.4
2024/03/07 21:10:16 [notice] 1#1: built by gcc 12.2.0 (Debian 12.2.0-14) 
2024/03/07 21:10:16 [notice] 1#1: OS: Linux 5.15.49-linuxkit
2024/03/07 21:10:16 [notice] 1#1: getrlimit(RLIMIT_NOFILE): 1048576:1048576
2024/03/07 21:10:16 [notice] 1#1: start worker processes
2024/03/07 21:10:16 [notice] 1#1: start worker process 29
2024/03/07 21:10:16 [notice] 1#1: start worker process 30
2024/03/07 21:10:16 [notice] 1#1: start worker process 31
2024/03/07 21:10:16 [notice] 1#1: start worker process 32
2024/03/07 21:10:16 [notice] 1#1: start worker process 33
2024/03/07 21:10:16 [notice] 1#1: start worker process 34
/docker-entrypoint.sh: /docker-entrypoint.d/ is not empty, will attempt to perform configuration
/docker-entrypoint.sh: Looking for shell scripts in /docker-entrypoint.d/
/docker-entrypoint.sh: Launching /docker-entrypoint.d/10-listen-on-ipv6-by-default.sh
10-listen-on-ipv6-by-default.sh: info: Getting the checksum of /etc/nginx/conf.d/default.conf
10-listen-on-ipv6-by-default.sh: info: Enabled listen on IPv6 in /etc/nginx/conf.d/default.conf
/docker-entrypoint.sh: Sourcing /docker-entrypoint.d/15-local-resolvers.envsh
/docker-entrypoint.sh: Launching /docker-entrypoint.d/20-envsubst-on-templates.sh
/docker-entrypoint.sh: Launching /docker-entrypoint.d/30-tune-worker-processes.sh
/docker-entrypoint.sh: Configuration complete; ready for start up
2024/03/07 21:15:54 [notice] 1#1: using the "epoll" event method
2024/03/07 21:15:54 [notice] 1#1: nginx/1.25.4
2024/03/07 21:15:54 [notice] 1#1: built by gcc 12.2.0 (Debian 12.2.0-14) 
2024/03/07 21:15:54 [notice] 1#1: OS: Linux 5.15.49-linuxkit
2024/03/07 21:15:54 [notice] 1#1: getrlimit(RLIMIT_NOFILE): 1048576:1048576
2024/03/07 21:15:54 [notice] 1#1: start worker processes
2024/03/07 21:15:54 [notice] 1#1: start worker process 29
2024/03/07 21:15:54 [notice] 1#1: start worker process 30
2024/03/07 21:15:54 [notice] 1#1: start worker process 31
2024/03/07 21:15:54 [notice] 1#1: start worker process 32
2024/03/07 21:15:54 [notice] 1#1: start worker process 33
2024/03/07 21:15:54 [notice] 1#1: start worker process 34
/docker-entrypoint.sh: /docker-entrypoint.d/ is not empty, will attempt to perform configuration
/docker-entrypoint.sh: Looking for shell scripts in /docker-entrypoint.d/
/docker-entrypoint.sh: Launching /docker-entrypoint.d/10-listen-on-ipv6-by-default.sh
10-listen-on-ipv6-by-default.sh: info: Getting the checksum of /etc/nginx/conf.d/default.conf
10-listen-on-ipv6-by-default.sh: info: Enabled listen on IPv6 in /etc/nginx/conf.d/default.conf
/docker-entrypoint.sh: Sourcing /docker-entrypoint.d/15-local-resolvers.envsh
/docker-entrypoint.sh: Launching /docker-entrypoint.d/20-envsubst-on-templates.sh
/docker-entrypoint.sh: Launching /docker-entrypoint.d/30-tune-worker-processes.sh
/docker-entrypoint.sh: Configuration complete; ready for start up
2024/03/07 21:15:54 [notice] 1#1: using the "epoll" event method
2024/03/07 21:15:54 [notice] 1#1: nginx/1.25.4
2024/03/07 21:15:54 [notice] 1#1: built by gcc 12.2.0 (Debian 12.2.0-14) 
2024/03/07 21:15:54 [notice] 1#1: OS: Linux 5.15.49-linuxkit
2024/03/07 21:15:54 [notice] 1#1: getrlimit(RLIMIT_NOFILE): 1048576:1048576
2024/03/07 21:15:54 [notice] 1#1: start worker processes
2024/03/07 21:15:54 [notice] 1#1: start worker process 29
2024/03/07 21:15:54 [notice] 1#1: start worker process 30
2024/03/07 21:15:54 [notice] 1#1: start worker process 31
2024/03/07 21:15:54 [notice] 1#1: start worker process 32
2024/03/07 21:15:54 [notice] 1#1: start worker process 33
2024/03/07 21:15:54 [notice] 1#1: start worker process 34
┌─[cmwylie19@Cases-MacBook-Pro] - [~/kubernetes] - [2024-03-11 06:38:24]
└─[0] <git:(1520 570a1ede3fd✱) > go run cmd/kubectl/kubectl.go logs deploy/y --all-pods --prefix
[pod/y-844fbc9777-vhl79/nginx] /docker-entrypoint.sh: /docker-entrypoint.d/ is not empty, will attempt to perform configuration
[pod/y-844fbc9777-vhl79/nginx] /docker-entrypoint.sh: Looking for shell scripts in /docker-entrypoint.d/
[pod/y-844fbc9777-vhl79/nginx] /docker-entrypoint.sh: Launching /docker-entrypoint.d/10-listen-on-ipv6-by-default.sh
[pod/y-844fbc9777-vhl79/nginx] 10-listen-on-ipv6-by-default.sh: info: Getting the checksum of /etc/nginx/conf.d/default.conf
[pod/y-844fbc9777-vhl79/nginx] 10-listen-on-ipv6-by-default.sh: info: Enabled listen on IPv6 in /etc/nginx/conf.d/default.conf
[pod/y-844fbc9777-vhl79/nginx] /docker-entrypoint.sh: Sourcing /docker-entrypoint.d/15-local-resolvers.envsh
[pod/y-844fbc9777-vhl79/nginx] /docker-entrypoint.sh: Launching /docker-entrypoint.d/20-envsubst-on-templates.sh
[pod/y-844fbc9777-vhl79/nginx] /docker-entrypoint.sh: Launching /docker-entrypoint.d/30-tune-worker-processes.sh
[pod/y-844fbc9777-vhl79/nginx] /docker-entrypoint.sh: Configuration complete; ready for start up
[pod/y-844fbc9777-vhl79/nginx] 2024/03/07 21:10:16 [notice] 1#1: using the "epoll" event method
[pod/y-844fbc9777-vhl79/nginx] 2024/03/07 21:10:16 [notice] 1#1: nginx/1.25.4
[pod/y-844fbc9777-vhl79/nginx] 2024/03/07 21:10:16 [notice] 1#1: built by gcc 12.2.0 (Debian 12.2.0-14) 
[pod/y-844fbc9777-vhl79/nginx] 2024/03/07 21:10:16 [notice] 1#1: OS: Linux 5.15.49-linuxkit
[pod/y-844fbc9777-vhl79/nginx] 2024/03/07 21:10:16 [notice] 1#1: getrlimit(RLIMIT_NOFILE): 1048576:1048576
[pod/y-844fbc9777-vhl79/nginx] 2024/03/07 21:10:16 [notice] 1#1: start worker processes
[pod/y-844fbc9777-vhl79/nginx] 2024/03/07 21:10:16 [notice] 1#1: start worker process 29
[pod/y-844fbc9777-vhl79/nginx] 2024/03/07 21:10:16 [notice] 1#1: start worker process 30
[pod/y-844fbc9777-vhl79/nginx] 2024/03/07 21:10:16 [notice] 1#1: start worker process 31
[pod/y-844fbc9777-vhl79/nginx] 2024/03/07 21:10:16 [notice] 1#1: start worker process 32
[pod/y-844fbc9777-vhl79/nginx] 2024/03/07 21:10:16 [notice] 1#1: start worker process 33
[pod/y-844fbc9777-vhl79/nginx] 2024/03/07 21:10:16 [notice] 1#1: start worker process 34
[pod/y-844fbc9777-hz4x4/nginx] /docker-entrypoint.sh: /docker-entrypoint.d/ is not empty, will attempt to perform configuration
[pod/y-844fbc9777-hz4x4/nginx] /docker-entrypoint.sh: Looking for shell scripts in /docker-entrypoint.d/
[pod/y-844fbc9777-hz4x4/nginx] /docker-entrypoint.sh: Launching /docker-entrypoint.d/10-listen-on-ipv6-by-default.sh
[pod/y-844fbc9777-hz4x4/nginx] 10-listen-on-ipv6-by-default.sh: info: Getting the checksum of /etc/nginx/conf.d/default.conf
[pod/y-844fbc9777-hz4x4/nginx] 10-listen-on-ipv6-by-default.sh: info: Enabled listen on IPv6 in /etc/nginx/conf.d/default.conf
[pod/y-844fbc9777-hz4x4/nginx] /docker-entrypoint.sh: Sourcing /docker-entrypoint.d/15-local-resolvers.envsh
[pod/y-844fbc9777-hz4x4/nginx] /docker-entrypoint.sh: Launching /docker-entrypoint.d/20-envsubst-on-templates.sh
[pod/y-844fbc9777-hz4x4/nginx] /docker-entrypoint.sh: Launching /docker-entrypoint.d/30-tune-worker-processes.sh
[pod/y-844fbc9777-hz4x4/nginx] /docker-entrypoint.sh: Configuration complete; ready for start up
[pod/y-844fbc9777-hz4x4/nginx] 2024/03/07 21:15:54 [notice] 1#1: using the "epoll" event method
[pod/y-844fbc9777-hz4x4/nginx] 2024/03/07 21:15:54 [notice] 1#1: nginx/1.25.4
[pod/y-844fbc9777-hz4x4/nginx] 2024/03/07 21:15:54 [notice] 1#1: built by gcc 12.2.0 (Debian 12.2.0-14) 
[pod/y-844fbc9777-hz4x4/nginx] 2024/03/07 21:15:54 [notice] 1#1: OS: Linux 5.15.49-linuxkit
[pod/y-844fbc9777-hz4x4/nginx] 2024/03/07 21:15:54 [notice] 1#1: getrlimit(RLIMIT_NOFILE): 1048576:1048576
[pod/y-844fbc9777-hz4x4/nginx] 2024/03/07 21:15:54 [notice] 1#1: start worker processes
[pod/y-844fbc9777-hz4x4/nginx] 2024/03/07 21:15:54 [notice] 1#1: start worker process 29
[pod/y-844fbc9777-hz4x4/nginx] 2024/03/07 21:15:54 [notice] 1#1: start worker process 30
[pod/y-844fbc9777-hz4x4/nginx] 2024/03/07 21:15:54 [notice] 1#1: start worker process 31
[pod/y-844fbc9777-hz4x4/nginx] 2024/03/07 21:15:54 [notice] 1#1: start worker process 32
[pod/y-844fbc9777-hz4x4/nginx] 2024/03/07 21:15:54 [notice] 1#1: start worker process 33
[pod/y-844fbc9777-hz4x4/nginx] 2024/03/07 21:15:54 [notice] 1#1: start worker process 34
[pod/y-844fbc9777-95rhd/nginx] /docker-entrypoint.sh: /docker-entrypoint.d/ is not empty, will attempt to perform configuration
[pod/y-844fbc9777-95rhd/nginx] /docker-entrypoint.sh: Looking for shell scripts in /docker-entrypoint.d/
[pod/y-844fbc9777-95rhd/nginx] /docker-entrypoint.sh: Launching /docker-entrypoint.d/10-listen-on-ipv6-by-default.sh
[pod/y-844fbc9777-95rhd/nginx] 10-listen-on-ipv6-by-default.sh: info: Getting the checksum of /etc/nginx/conf.d/default.conf
[pod/y-844fbc9777-95rhd/nginx] 10-listen-on-ipv6-by-default.sh: info: Enabled listen on IPv6 in /etc/nginx/conf.d/default.conf
[pod/y-844fbc9777-95rhd/nginx] /docker-entrypoint.sh: Sourcing /docker-entrypoint.d/15-local-resolvers.envsh
[pod/y-844fbc9777-95rhd/nginx] /docker-entrypoint.sh: Launching /docker-entrypoint.d/20-envsubst-on-templates.sh
[pod/y-844fbc9777-95rhd/nginx] /docker-entrypoint.sh: Launching /docker-entrypoint.d/30-tune-worker-processes.sh
[pod/y-844fbc9777-95rhd/nginx] /docker-entrypoint.sh: Configuration complete; ready for start up
[pod/y-844fbc9777-95rhd/nginx] 2024/03/07 21:15:54 [notice] 1#1: using the "epoll" event method
[pod/y-844fbc9777-95rhd/nginx] 2024/03/07 21:15:54 [notice] 1#1: nginx/1.25.4
[pod/y-844fbc9777-95rhd/nginx] 2024/03/07 21:15:54 [notice] 1#1: built by gcc 12.2.0 (Debian 12.2.0-14) 
[pod/y-844fbc9777-95rhd/nginx] 2024/03/07 21:15:54 [notice] 1#1: OS: Linux 5.15.49-linuxkit
[pod/y-844fbc9777-95rhd/nginx] 2024/03/07 21:15:54 [notice] 1#1: getrlimit(RLIMIT_NOFILE): 1048576:1048576
[pod/y-844fbc9777-95rhd/nginx] 2024/03/07 21:15:54 [notice] 1#1: start worker processes
[pod/y-844fbc9777-95rhd/nginx] 2024/03/07 21:15:54 [notice] 1#1: start worker process 29
[pod/y-844fbc9777-95rhd/nginx] 2024/03/07 21:15:54 [notice] 1#1: start worker process 30
[pod/y-844fbc9777-95rhd/nginx] 2024/03/07 21:15:54 [notice] 1#1: start worker process 31
[pod/y-844fbc9777-95rhd/nginx] 2024/03/07 21:15:54 [notice] 1#1: start worker process 32
[pod/y-844fbc9777-95rhd/nginx] 2024/03/07 21:15:54 [notice] 1#1: start worker process 33
[pod/y-844fbc9777-95rhd/nginx] 2024/03/07 21:15:54 [notice] 1#1: start worker process 34
```

### CODE REVIEW

File: staging/src/k8s.io/kubectl/pkg/cmd/logs/logs.go [hunk 1]
```diff
@@ -134,10 +136,9 @@ type LogsOptions struct {
 	containerNameFromRefSpecRegexp *regexp.Regexp
 }
 
-func NewLogsOptions(streams genericiooptions.IOStreams, allContainers bool) *LogsOptions {
+func NewLogsOptions(streams genericiooptions.IOStreams) *LogsOptions {
 	return &LogsOptions{
```
Comment (wangyysde) [inline staging/src/k8s.io/kubectl/pkg/cmd/logs/logs.go:140]: could you tell me why remove "allContainers"?

File: staging/src/k8s.io/kubectl/pkg/cmd/logs/logs.go [hunk 2]
```diff
@@ -167,6 +168,7 @@ func NewCmdLogs(f cmdutil.Factory, streams genericiooptions.IOStreams) *cobra.Co
 }
 
 func (o *LogsOptions) AddFlags(cmd *cobra.Command) {
+	cmd.Flags().BoolVar(&o.AllPods, "all-pods", o.AllPods, "Get logs from all pod(s). Sets prefix to true.")
 	cmd.Flags().BoolVar(&o.AllContainers, "all-containers", o.AllContainers, "Get all containers' logs in the pod(s).")
```
Comment (wangyysde) [inline staging/src/k8s.io/kubectl/pkg/cmd/logs/logs.go:172]: parameter named "allContainers" in the function "NewLogsOptions" has be removed.

File: staging/src/k8s.io/kubectl/pkg/cmd/logs/logs.go (hunk 1 above)
Comment (eddiezane) [reply staging/src/k8s.io/kubectl/pkg/cmd/logs/logs.go:140]: There's some context on the previous iteration of this PR.

https://github.com/kubernetes/kubernetes/pull/123874#discussion_r1548680888

TLDR: It shouldn't have been added to the signature in the first place.

File: staging/src/k8s.io/kubectl/pkg/cmd/logs/logs.go (hunk 2 above)
Comment (eddiezane) [reply staging/src/k8s.io/kubectl/pkg/cmd/logs/logs.go:172]: I am not sure I understand?

The flag `--all-containers` was not removed, just the parameter from the constructor.

**Comment by eddiezane** [comment]
Aside from the linter, the tests failures are legitimate timeouts. [0](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/124732/pull-kubernetes-e2e-kind-ipv6/1787992581041295360) [1](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/124732/pull-kubernetes-e2e-kind/1787992580974186496) [2](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/124732/pull-kubernetes-e2e-gce/1787992580768665600) . Need to investigate them a bit deeper tomorrow.

**Comment by eddiezane** [comment]
@aojea any idea why the tests would be failing like this? The test runs fine locally but is hitting the 10 minute timeout while trying to spin up a StatefulSet. It looks like it manages to get 1 pod up at one point but then falls back to 0.

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/124732/pull-kubernetes-e2e-kind/1788771572169838592#1:build-log.txt%3A667-741

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/124732/pull-kubernetes-e2e-gce/1788771572048203776#1:build-log.txt%3A781-855

**Comment by eddiezane** [comment]
It looks like both StatefulSets were actually created and logged out successfully

https://gcsweb.k8s.io/gcs/kubernetes-jenkins/pr-logs/pull/124732/pull-kubernetes-e2e-kind/1788771572169838592/artifacts/kubectl-logs-2956/

File: test/e2e/kubectl/logs.go [hunk 3]
```diff
@@ -33,13 +38,55 @@ import (
...
+							Image: imageutils.GetE2EImage(imageutils.Agnhost),
+							Args:  []string{"logs-generator", "--log-lines-total", "20", "--run-duration", "5s"},
+						},
+					},
+					RestartPolicy: v1.RestartPolicyAlways,
+				},
+			},
+		},
+	}
+}
```
Comment (eddiezane) [inline test/e2e/kubectl/logs.go:89]: nit: new line

File: test/e2e/kubectl/logs.go [hunk 4]
```diff
@@ -33,13 +38,55 @@ import (
...
+							Image: imageutils.GetE2EImage(imageutils.Agnhost),
+							Args:  []string{"logs-generator", "--log-lines-total", "10", "--run-duration", "5s"},
+						},
+						{
+							Name:  "container-2",
+							Image: imageutils.GetE2EImage(imageutils.Agnhost),
+							Args:  []string{"logs-generator", "--log-lines-total", "20", "--run-duration", "5s"},
+						},
+					},
+					RestartPolicy: v1.RestartPolicyAlways,
```
Comment (eddiezane) [inline test/e2e/kubectl/logs.go:84]: This might be the actual cause of the failures. Let's try setting this to never. What might be happening is that the pods are execting before they can be succesfully recongized as started by the wait.

**Review by eddiezane — CHANGES_REQUESTED** [review]

File: test/e2e/kubectl/logs.go (hunk 4 above)
Comment (cmwylie19) [reply test/e2e/kubectl/logs.go:84]: Unfortunately seems as though that is invalid

```bash
 [FAILED] Failed to create StatefulSet: StatefulSet.apps "sts7c9438e1-d736-4c2a-ae3e-cbb9ef945d32" is invalid: spec.template.spec.restartPolicy: Unsupported value: "Never": supported values: "Always"
  In [BeforeEach] at: k8s.io/kubernetes/test/e2e/kubectl/logs.go:274
```

File: test/e2e/kubectl/logs.go [hunk 5]
```diff
@@ -209,4 +257,100 @@ var _ = SIGDescribe("Kubectl logs", func() {
...
+							deployPodContainer = true
+						}
+						gomega.Expect(deployPodContainer).To(gomega.BeTrueBecause("each log should be from all containers from all pods in the Deployment"))
+					}
+				}
+				gomega.Expect(logFound).To(gomega.BeTrueBecause("log should be present"))
+				gomega.Expect(strings.Contains(out, fmt.Sprintf("[pod/%s/container-1]", podOne))).To(gomega.BeTrueBecause("pod 1 container 1 log should be present"))
+				gomega.Expect(strings.Contains(out, fmt.Sprintf("[pod/%s/container-2]", podOne))).To(gomega.BeTrueBecause("pod 1 container 2 log should be present"))
+				gomega.Expect(strings.Contains(out, fmt.Sprintf("[pod/%s/container-1]", podTwo))).To(gomega.BeTrueBecause("pod 1 container 1 log should be present"))
+				gomega.Expect(strings.Contains(out, fmt.Sprintf("[pod/%s/container-2]", podTwo))).To(gomega.BeTrueBecause("pod 1 container 2 log should be present"))
```
Comment (eddiezane) [inline test/e2e/kubectl/logs.go:348]: I think this should say `pod 2`.

**Review by eddiezane — CHANGES_REQUESTED** [review]
One small change then lgtm!

/approve

File: test/e2e/kubectl/logs.go (hunk 5 above)
Comment (cmwylie19) [reply test/e2e/kubectl/logs.go]: haha yes, great eye, and attention to detail. Thanks!

**Review by eddiezane — APPROVED** [review]
Awesome work on this!

/lgtm

**Comment by BenTheElder** [comment]
the e2e test seems flaky?

https://go.k8s.io/triage?test=Kubectl%20logs%20all%20pod%20logs%20the%20Deployment%20has%202%20replicas%20

example: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/125035/pull-kubernetes-e2e-kind-ipv6/1793029056485134336

**Comment by BenTheElder** [comment]
(this happens _all the time_, making e2e tests reliable is hard, but if I'm right and it is this test it would be appreciated to investigate and attempt to increase e2e test reliability 🙏 , at a quick glance nothing jumps out in the test code)

**Comment by eddiezane** [comment]
Spoke with Ben. Going to submit something.
